### DISCUSSION TITLE
migration of cudaMemcpyToSymbol

### DISCUSSION THREAD

**Discussion by zjin-lcf** [intel/llvm#7501] [Q&A]
Could you please advise how to map cudaMemcpyToSymbol() in CUDA to SYCL USM ?  Thanks.

**Comment by JackAKirk** [comment]
This question is relevant for the current work on using constant symbols. Basically I think that this functionality isn't fully supported yet but it is being worked on. We may have to consider whether we want something that maps directly to `cudaMemcpyToSymbol()` or just `cudaGetSymbolAddress` I guess.  CC @jchlanda

**Reply by zjin-lcf** [reply]
The USM allocation functions with properties support  (https://github.com/intel/llvm/pull/6346) may be relevant.

I assume that users  first allocate a memory by calling the function malloc_device(). The function accepts an attribute for the constant memory space.
Then, users copy data from a host to this allocated memory with q.memcpy(). 
Finally, the memory is accessed like other global memory in a SYCL kernel.

In a CUDA program, a CUDA kernel could access a constant memory without passing it as a kernel argument.  In some CUDA applications, there are many constant memories, so this improves productivity at least.
